appcues/src/main/java/com/appcues/debugger/screencapture/SaveCaptureUseCase.kt

```diff
@@ -0,0 +1,86 @@
+package com.appcues.debugger.screencapture
+
+import android.graphics.Bitmap
+import com.appcues.data.remote.RemoteError
+import com.appcues.data.remote.customerapi.CustomerApiRemoteSource
+import com.appcues.data.remote.imageupload.ImageUploadRemoteSource
+import com.appcues.data.remote.sdksettings.SdkSettingsRemoteSource
+import com.appcues.debugger.screencapture.model.Capture
+import com.appcues.util.ResultOf
+import com.appcues.util.ResultOf.Failure
+import com.appcues.util.ResultOf.Success
+
+internal class SaveCaptureUseCase(
+    private val sdkSettings: SdkSettingsRemoteSource,
+    private val customer: CustomerApiRemoteSource,
+    private val imageUpload: ImageUploadRemoteSource,
+) {
+
+    suspend operator fun invoke(token: String, capture: Capture): ResultOf<Unit, RemoteError> {
+        // Saving a screen is a 4-step chain. This is implemented here as a sequence of calls, chaining
+        // a success continuation on each call to move to the next step. If any step fails, the RemoteError
```

Comment:
this may be an old comment, a little outdated with the wording around "chaining a success continuation..."